src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -75,6 +77,79 @@ public class BlobFileStore : IFileStore
         }
     }
 
+    /// <inheritdoc />
+    public async Task<Uri> StoreFileInBlocksAsync(
+        long version,
+        Stream stream,
+        IDictionary<string, long> blockLengths,
+        CancellationToken cancellationToken)
+    {
+        EnsureArg.IsNotNull(stream, nameof(stream));
+        EnsureArg.IsNotNull(blockLengths, nameof(blockLengths));
+        EnsureArg.IsGte(blockLengths.Count, 0, nameof(blockLengths.Count));
+
+        BlockBlobClient blobClient = GetInstanceBlockBlobClient(version);
+
+        try
+        {
+            stream.Seek(0, SeekOrigin.Begin);
+
+            foreach (string blockId in blockLengths.Keys)
+            {
+                long blockSize = blockLengths[blockId];
+                byte[] blockData = new byte[blockSize];
```

Comment:
Instead of creating an array, simply rent an array from an [`ArrayPool<T>`](https://learn.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1?view=net-7.0). Then you can read into the buffer and create a stream from it.